src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs

```diff
@@ -163,104 +163,138 @@ public void TestDatabaseRestore()
+                sqlClient.GetPutOrPatchOperationResultAsync(policyResult, new Dictionary<string, List<string>>(), CancellationToken.None).Wait();
+
+                // Get the policy and verify the weekly policy is two weeks
+                //
+                Microsoft.Azure.Management.Sql.Models.BackupLongTermRetentionPolicy policy = sqlClient.BackupLongTermRetentionPolicies.Get(resourceGroupName, serverName, databaseName);
+                Assert.Equal(parameters.WeeklyRetention, policy.WeeklyRetention);
+
+                // Get the backups under the location, server, and database. Assert there is at least one backup for each call.
+                //
+                IPage<LongTermRetentionBackup> backups = sqlClient.LongTermRetentionBackups.ListByLocation(locationName);
+                Assert.True(backups.Count() >= 1);
+                backups = sqlClient.LongTermRetentionBackups.ListByServer(locationName, serverName);
+                Assert.True(backups.Count() >= 1);
+                backups = sqlClient.LongTermRetentionBackups.ListByDatabase(locationName, serverName, databaseName);
+                Assert.True(backups.Count() >= 1);
+
+                // Get a specific backup using the previous call
+                //
+                LongTermRetentionBackup backup = sqlClient.LongTermRetentionBackups.Get(locationName, serverName, databaseName, backups.First().Name);
+                Assert.NotNull(backup);
+
+                // Restore the backup - TODO After Database 2017-03-01-Preview is done
+                //
+                /* Database restoredDatabase = sqlClient.Databases.CreateOrUpdate(
+                    resourceGroupName, serverName, databaseName: SqlManagementTestUtilities.GenerateName(),
                     parameters: new Database
                     {
-                        Location = resourceGroup.Location,
+                        Location = locationName,
                         CreateMode = CreateMode.RestoreLongTermRetentionBackup,
-                        RecoveryServicesRecoveryPointResourceId = // recovery point resource id
-                    });
-                */
+                        RecoveryServicesRecoveryPointResourceId = ""
+                    });*/
+
+                // Delete the backup.
+                //
+                var deleteResult = sqlClient.LongTermRetentionBackups.BeginDeleteWithHttpMessagesAsync(locationName, serverName, databaseName, backup.Name).Result;
+                sqlClient.GetPutOrPatchOperationResultAsync(deleteResult, new Dictionary<string, List<string>>(), CancellationToken.None).Wait();
+
+                // Verify the backup is gone.
+                //
+                try
+                {
+                    sqlClient.LongTermRetentionBackups.Get(locationName, serverName, databaseName, backup.Name);
```

Comment:
you are not actually asserting that the Get failed. use `CloudException ex = Assert.Throws<CloudException>(..); Assert.AreEqual(ex.Response.Code, HttpStatusCode.NotFound);`